components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -4767,114 +4757,128 @@ end subroutine mam_coag_1subarea
 ! *** need to add qaer_del_coag_nmoacc_in
-      real(r8), intent(inout), dimension( 1:max_mode ) :: &
-         qwtr_cur
+
 
 ! local variables
-      integer :: iaer, ipair, itmpa
-      integer :: nfrm, ntoo
+      integer :: iaer, ipair
+      integer :: nsrc, ndest
 
-      real(r8) :: fac_volsfc
-      real(r8) :: tmpa, tmp1, tmp2, tmp3, tmp4
-      real(r8) :: vol_core, vol_shell
-      real(r8) :: xferfrac_max, xferfrac_pcage
+      real(r8) :: xferfrac_pcage, frac_cond, frac_coag
 
 
 ! 
 agepair_loop1: &
-      do ipair = 1, n_agepair
+      do ipair = 1, n_agepair         ! only execute for ipair = 1, ipair>=2 is for MAM9
 
-      nfrm = modefrm_agepair(ipair)
-      ntoo = modetoo_agepair(ipair)
+      nsrc  = modefrm_agepair(ipair)
+      ndest = modetoo_agepair(ipair)
 
-      vol_shell = qaer_cur(iaer_so4,nfrm)*fac_m2v_aer(iaer_so4)
-      tmp3 = qaer_del_cond(iaer_so4,nfrm)  *fac_m2v_aer(iaer_so4)
-      tmp4 = qaer_del_coag_in(iaer_so4,ipair)*fac_m2v_aer(iaer_so4)
+      call mam_pcarbon_aging_frac(nsrc, ipair, dgn_a,  &  ! input
+            qaer_cur, qaer_del_cond, qaer_del_coag_in, &  ! in-outs
+            xferfrac_pcage, frac_cond, frac_coag)         ! output
 
       do iaer = 1, naer
-!     species that contribute to aging are 
-!        so4 (but it is already done)
-!        soa, nh4 and no3
-!        ncl and cl (when aging_include_seasalt == .true.)
-         if ( (iaer <= nsoa    ) .or. &
-              (iaer == iaer_nh4) .or. &
-              (iaer == iaer_no3) .or. &
-              (iaer == iaer_cl ) ) then
-            continue
-         else if (iaer == iaer_ncl) then
-            if (aging_include_seasalt .eqv. .false.) cycle
+         if (lmap_aer(iaer,nsrc) > 0) then   ! MAM4 pcarbon mode only has pom, bc, mom, lmap only has index (>0) for these species
+            ! species is pom or bc
+            ! transfer the aged fraction to accum mode
+            ! include this transfer change in the cond and/or coag change (for mass budget)
+            call transfer_aged_pcarbon_to_accum(nsrc, ndest, &                                                  ! input
+                                                xferfrac_pcage, frac_cond, frac_coag, &                         ! input
+                                                qaer_cur(iaer,:), qaer_del_cond(iaer,:), qaer_del_coag(iaer,:)) ! in-outs
          else
-            cycle
-         end if
+            ! species is soa, so4, or nh4 produced by condensation or coagulation
+            ! transfer all of it to accum mode
+            ! also transfer the condensation and coagulation changes
+            !    to accum mode (for mass budget)
+            call transfer_cond_coag_mass_to_accum(nsrc, ndest, &                                                  ! input
+                                                  qaer_cur(iaer,:), qaer_del_cond(iaer,:), qaer_del_coag(iaer,:)) ! in-outs
```

Comment:
I am unsure if calling this subroutine using `(iaer,:)` is helpful as it is always called with the same inputs. It is no big deal, we can leave it as is.

Reply:
Thank you. I was thinking to be consistent with transfer_aged_pcarbon_to _accum. Maybe it could save some memory/be faster when operating one dimension array than two dimension array.

Reply:
In this case, it shouldn't matter much as referencing arrays of known indices are all O(1) operations if the arrays are read contiguously (i.e. column-wise). One side effect of using array slicing (`(iaer,:)`) is that it forces the compiler to create temporary arrays on the fly. 
On the plus side, it explicitly shows which dimension we are operating on in these subroutine calls.

Reply:
The overhead of passing array slices might not be too bad in all cases, actually: 
https://stackoverflow.com/questions/29088071/fortran-performance-when-passing-array-slices-as-arguments

Clarity is a big benefit. If we are concerned about the performance overhead, we could measure it.